nym-vpn-core/crates/nym-statistics/src/api_client.rs

```diff
@@ -0,0 +1,38 @@
+// Copyright 2025 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: GPL-3.0-only
+
+use serde::Serialize;
+
+use crate::config::StatisticsControllerConfig;
+use crate::error::Error;
+
+#[derive(Clone, Debug)]
+pub(crate) struct StatisticsControllerApiClient {
+    inner: nym_statistics_api_client::StatisticsApiClient,
+}
+
+impl StatisticsControllerApiClient {
+    pub(crate) fn new(config: &StatisticsControllerConfig) -> Result<Option<Self>, Error> {
```

Comment:
Nit: just a personal opinion. Since the struct is already `pub(crate)` you can save yourself from repeating `pub(crate)` on individual methods, `pub` should work just fine and it's shorter too.